test/unit/app/browser/isThirdPartyhostTest.js

```diff
@@ -12,11 +12,40 @@ describe('isThirdPartyHost test', function () {
   })
   it('A subdomain URL should not be third party', function () {
     assert.ok(!isThirdPartyHost(braveHost, 'ragu.brave.com'))
+    assert.ok(!isThirdPartyHost('ragu.brave.com', braveHost))
+  })
+  it('A 2nd level subdomain URL should not be third party', function () {
+    assert.ok(!isThirdPartyHost(braveHost, 'foo.bar.brave.com'))
+    assert.ok(!isThirdPartyHost('foo.bar.brave.com', braveHost))
   })
   it('Unrelated URLs should be third party', function () {
     assert.ok(isThirdPartyHost(braveHost, 'ragu.com'))
+    assert.ok(isThirdPartyHost('ragu.com', braveHost))
   })
   it('Checks subdomains properly', function () {
     assert.ok(isThirdPartyHost(braveHost, 'brave.ragu.com'))
+    assert.ok(isThirdPartyHost('brave.ragu.com', braveHost))
+  })
+  it('Handles multi-part TLDs', function () {
+    assert.ok(isThirdPartyHost('diracdeltas.github.io', 'brave.github.io'))
+    assert.ok(isThirdPartyHost('github.io', 'brave.github.io'))
+    assert.ok(!isThirdPartyHost('github.io', 'github.io'))
+    assert.ok(isThirdPartyHost('brave.github.io', 'github.io'))
+    assert.ok(isThirdPartyHost('brave.co.uk', 'example.co.uk'))
+  })
+  it('Handles IPv4', function () {
+    assert.ok(isThirdPartyHost('172.217.6.46', '173.217.6.46'))
+    assert.ok(!isThirdPartyHost('172.217.6.46', '172.217.6.46'))
```

Comment:
I think it's clear from https://github.com/brave/browser-laptop/pull/13820/files#diff-3e19062054041e338c28b2593b82e8d9R23 that this is not intended to handle IP representations that are equivalent but don't have the same string representation for now

Reply:
or are you suggesting adding a test for the equivalent case returning third party so that it's explicit based on tests what the behavior should be? i can do that

Reply:
^ probably best to do in a follow-up PR since it's non-critical and adding it here would dismiss the existing reviews